### DISCUSSION TITLE
How to get binary string in js

### DISCUSSION THREAD

**Discussion by atlanticmr** [community/community#68852] [Programming Help] [answered]
### Body

Which way to work with bytes & bits in JS

### Guidelines

- [x] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by yarscript** [comment, marked as answer]
JavaScript doesn't provide native capabilities to work with bits and bytes directly like some other programming languages. However, you can use some built-in functions and operators to work with binary data at the bit level. 
 
For bitwise operations, JavaScript provides bitwise operators: 
 
- Bitwise AND (&) 
- Bitwise OR (|) 
- Bitwise XOR (^) 
- Bitwise NOT (~) 
- Left shift (<<) 
- Sign-propagating right shift (>>) 
- Zero-fill right shift (>>>) 
 
For example:
```js
let a = 5;      // binary: 0101
let b = 3;      // binary: 0011

console.log(a & b);  // binary: 0001, decimal: 1
console.log(a | b);  // binary: 0111, decimal: 7
console.log(a ^ b);  // binary: 0110, decimal: 6
console.log(~a);     // binary: 1010, decimal: -6
console.log(a << 1); // binary: 1010, decimal: 10
console.log(a >> 1); // binary: 0010, decimal: 2
console.log(a >>> 1); // binary: 0010, decimal: 2
```

For working with bytes, you can use  ArrayBuffer  and typed arrays ( Uint8Array ,  Int16Array , etc.).  ArrayBuffer  represents a generic, fixed-length raw binary data buffer, and typed arrays provide a way to access the bytes of the buffer as specific types. 
 
Here is an example of using  ArrayBuffer  and  Uint8Array :
```js
let buffer = new ArrayBuffer(8);  // create an 8-byte buffer
let bytes = new Uint8Array(buffer);  // create a view for 8-bit unsigned integers

bytes[0] = 255;  // set the first byte to 255
console.log(bytes[0]);  // 255
```

**Reply by atlanticmr** [reply]
Thanks a lot!
It looks like exact what I needed 💪

**Reply by yarscript** [reply]
come again 😸

**Comment by OshandaB** [comment]
Response #3
